OneCommunityGlobal/HighestGoodNetworkApp
# Imran manage admin links in user profile

# Description
Gives the user permission to edit the links of any user on the user profile page. "User Profile" -> "Links button"

## Related PRS (if any):
This frontend PR is related to the [PR#916](https://github.com/OneCommunityGlobal/HGNRest/pull/916) backend PR.

## Main changes explained:
- Add manageAdminLinks to PermissionsConst.js
- replaced 'putUserProfileImportantInfo'  permission check with 'manageAdminLinks' in EditLinkModal.jsx

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as admin user -> Other Links -> Permissions Management -> Manage User Permissions.
5. Enter the username of an user that does not have the “Manage Admin Links in User Profile” permission. Add the permission and Click on Submit.
6. Login as the chosen user (if already logged in, make sure to log out and log back in). 
7. Navigate to  "User Profile" -> "Links" -> "Edit". The user should now be able to edit their admin links.
8. Visit any other user's profile and should be able to edit admin links.

## Screenshots or videos of changes:
**Navigate to Profile as a without permission to manage Admin Links. Verify that the edit button is visible** 
![Step 1](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/87269725/d02e6c43-28c5-4110-9d2a-c016b6fab7a5)

**Verify that you can't see admin Links**
![Step 2](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/87269725/081250d8-e364-46c5-aaf1-cb31c8482ab0)

**Visit profile of any other user and verify you can't see edit button next to Links. (Note: if the logged in user has other permissions such as Edit User Profile, you should be able to see the button but not admin links in the dialog**
![Step 3 (2)](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/87269725/4036dcaf-3f06-4031-99a0-e40039008a52)

**Logout and Log back in as admin and navigate to permission management screen to give user permission to manage admin links**
![Step 3](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/87269725/e9bcc87b-e2fc-4e86-b9f6-379805830cf8)
![Step 4](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/87269725/e21f3f27-99e4-46ba-b2cf-7f96ecd83642)

**Log back in as the user given permission and navigate to profile and verify you can now manage admin links**
![Step 5](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/87269725/c2302f4d-e478-4a3f-8520-4e56cc6c3158)
![Step 6](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/87269725/b4fee2b4-4ef5-423d-bd6a-89236ed8447e)

**Visit any other user's profile and verify you can now add or edit their admin links too**
![Step 7](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/87269725/67769079-8874-4ba5-83e2-5968c8bc7809)
